tensorflow/core/kernels/unravel_index_op.cc

```diff
@@ -54,6 +54,18 @@ class UnravelIndexOp : public OpKernel {
 
     auto dims = dims_tensor.vec<Tidx>();
 
+    // Chek to make sure indices is not out of boundary
+    Eigen::Tensor<Tidx, 0, Eigen::RowMajor> dims_prod_eigen = dims.prod();
+    Tidx dims_prod = dims_prod_eigen();
+    Tidx* indices = (Tidx*)(indices_tensor.flat<Tidx>().data());
```

Comment:
c-style casts are against style guide (https://google.github.io/styleguide/cppguide.html#Casting). Is this cast required to remove `const`? I think `const Tidx* indices` should also work just fine with `std:all_of`.

Reply:
@ezhulenev Thanks. The PR has been updated with C cast removed. Please take a look.